test-monitor/nitestmonitor.yml

```diff
@@ -0,0 +1,1277 @@
+        default:
+          $ref: '#/responses/Error'
+  /v1/steps:
+    put:
+      operationId: update-steps
+      x-ni-operation: updateSteps
+      x-ni-privilege: Write
+      summary: Update test steps
+      description: Update test steps
+      tags:
+        - steps
+      parameters:
+        - in: body
+          name: requestBody
+          description: A list of test steps
+          required: true
+          schema:
+            $ref: '#/definitions/TestStepCreateOrUpdateRequestObject'
+      responses:
+        200:
+          $ref: '#/responses/StepsPartialSuccessResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+    post:
+      operationId: create-steps
+      x-ni-operation: createSteps
+      x-ni-privilege: Write
+      summary: Create test steps
+      description: Create test steps
+      tags:
+        - steps
+      parameters:
+        - in: body
+          name: requestBody
+          description: A test step object
+          required: true
+          schema:
+            $ref: '#/definitions/TestStepCreateOrUpdateRequestObject'
+      responses:
+        200:
+          $ref: '#/responses/StepsPartialSuccessResponse'
+        201:
+          $ref: '#/responses/StepsSuccessResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/results/{resultId}/steps/{stepId}:
+    delete:
+      operationId: DeleteStep
+      x-ni-operation: deleteStep
+      x-ni-privilege: Write
+      summary: Deletes a test step
+      description: >-
+        Deletes a test step
+      tags: 
+       - steps
+      parameters:
+        - in: path
+          name: resultId
+          description: The id of the test result the step belongs to
+          type: string
+          required: true
+        - in: path
+          name: stepId
+          description: The id of the test step to delete
+          type: string
+          required: true
+      responses:
+        204:
+          description: The resource was deleted successfully.
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/delete-steps:
+    post:
+      operationId: delete-steps
+      x-ni-operation: deleteSteps
+      x-ni-privilege: Write
+      summary: Delete test steps
+      description: Delete test steps
+      tags:
+        - steps
+      parameters:
+        - in: body
+          name: requestBody
+          description: A delete steps request
+          required: true
+          schema:
+            $ref: '#/definitions/TestStepsDeleteRequest'
+      responses:
+        200:
+          $ref: '#/responses/StepsDeletePartialSuccessResponse'
+        204:
+          description: The resource was deleted successfully.
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/ping:
```

Comment:
Have we shipped documentation for this route to customers? If we haven't, I would think we can just leave this route undocumented, as it's deprecated.

Reply:
We are using codegen for this service, so it has to be documented unless we remove it from the service (which we probably should anyway).

Reply:
Should we note it's deprecated outside of the tags section, like in the description?

Reply:
I've marked it deprecated in the description, although I agree we should make any required changes to the UI and drop it from the API entirely.  I'll do that as a separate change since it requires a backend code change.